dotcom-rendering/src/lib/useAdBlockAsk.ts

```diff
@@ -81,14 +82,29 @@ export const useAdblockAsk = ({
 }): boolean => {
 	const isInVariant = useIsInAdBlockAskVariant();
 	const [adBlockerDetected, setAdBlockerDetected] = useState<boolean>(false);
-
-	const authStatus = useAuthStatus();
-	const isSignedIn =
-		authStatus.kind === 'SignedInWithOkta' ||
-		authStatus.kind === 'SignedInWithCookies';
+	const [isAdFree, setIsAdFree] = useState<boolean>(false);
+	const [hasConsentForGoogletag, setHasConsentForGoogletag] = useState(false);
 
 	const canDisplayAdBlockAsk =
-		!shouldHideReaderRevenue && !isPaidContent && !isSignedIn;
+		!shouldHideReaderRevenue &&
+		!isPaidContent &&
+		!isAdFree &&
+		hasConsentForGoogletag;
+
+	useEffect(() => {
+		onConsentChange((consentState) => {
+			if (consentState.tcfv2) {
+				setHasConsentForGoogletag(
+					getConsentFor('googletag', consentState),
+				);
+			}
+			setHasConsentForGoogletag(true);
```

Comment:
When `consentState.tcfv2` is true will both of the `setHasConsentForGoogletag` calls fire?

Reply:
ah thanks! missed the early return